src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -485,15 +527,39 @@ private void initializeReads(final JavaSparkContext sparkContext) {
             return;
         }
 
-        if ( readArguments.getReadFilesNames().size() != 1 ) {
-            throw new UserException("Sorry, we only support a single reads input for spark tools for now.");
+        if (getReadInputMergingPolicy() == ReadInputMergingPolicy.doNotMerge && readArguments.getReadFilesNames().size() != 1 ) {
+            throw new UserException("Sorry, we only support a single reads input for for this spark tool.");
         }
 
-        readInput = readArguments.getReadFilesNames().get(0);
+        readInputs = new LinkedHashMap<>();
         readsSource = new ReadsSparkSource(sparkContext, readArguments.getReadValidationStringency());
-        readsHeader = readsSource.getHeader(
-                readInput,
-                hasReference() ?  referenceArguments.getReferenceFileName() : null);
+        for (String input : readArguments.getReadFilesNames()) {
+            readInputs.put(input, readsSource.getHeader(
+                    input, hasReference() ?  referenceArguments.getReferenceFileName() : null));
+        }
+        readsHeader = createHeaderMerger().getMergedHeader();
+    }
+
+
+
+    /**
+     * Create a header merger from the individual SAM/BAM headers in our readers
+     *
+     * @return a header merger containing all individual headers in this data source
+     */
+    private SamFileHeaderMerger createHeaderMerger() {
+        return new SamFileHeaderMerger(identifySortOrder(readInputs.values()), readInputs.values(), true);
+    }
```

Comment:
There should be a newline here:
```suggestion
    }
    
```